### ISSUE TITLE
MemMapRouteDefiner and PipelineRouteDefiner should be extended from a base class

### ISSUE THREAD

**Issue by ShamsUlAzeem** [KonduitAI/konduit-serving#102]
With the addition of a few endpoints in this PR: https://github.com/KonduitAI/konduit-serving/pull/98, the memmap and pipeline route definers have some duplicate endpoints. So, to avoid code duplication we can either have a utility function that creates those route for us or we can have a base route definer and extend both of these classes from it.

**Comment by agibsonccc** [comment]
Hey folks I think we should look closer at this before defining abstractions.

@AlexDBlack has a route definition framework he built in the dl4j UI  for example. Let's consider a more general solution for the clean up before we do the refactoring here.

One thing would be allowing custom route definers that are looked up on the classpath.

**Comment by ShamsUlAzeem** [comment]
Linking this for reference: Thanks @AlexDBlack 

https://github.com/eclipse/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-ui-parent/deeplearning4j-vertx/src/main/java/org/deeplearning4j/ui/api/UIModule.java#L51-L56
https://github.com/eclipse/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-ui-parent/deeplearning4j-vertx/src/main/java/org/deeplearning4j/ui/VertxUIServer.java#L196-L210
basically, we discover instances of UIModule either directly or via Java ServiceLoader mechanism
then get the routes from each UIModule when constructing the server
basically, we define the routes dynamically so the UI is extensible without modifying the core server creation code
